### DISCUSSION TITLE
how do you add a number in a string?

### DISCUSSION THREAD

**Discussion by MrMidnightOfficial** [vlang/v#14885] [General]
how do you add a number in a string?

example:
`mut fat := "g" //Add g to the end of the string to repesent grams`
`food_add_fat_user_input := os.input('What do you want to add to Fat:') //Fat input`
`fat = food_add_fat_user_input + 'g' //Add g to the end of the string to repesent grams`

**Comment by JalonSolov** [comment]
Example?

**Reply by MrMidnightOfficial** [reply]
Yes

**Reply by Wajinn** [reply]
> Yes

There is possibly some language confusion going on.  Your original question creates some confusion about what you want or what you might mean.  I think he is asking you to provide an example or to provide more of an explanation.  Just typing, "yes", would not be helpful.

**Reply by JalonSolov** [reply]
Yes, I was asking for an example of what is being requested.

**Reply by MrMidnightOfficial** [reply]
> Yes, I was asking for an example of what is being requested.

Oh sorry I thot you were asking me if I want an example here is an example

`mut fat := "g" //Add g to the end of the string to repesent grams`
`food_add_fat_user_input := os.input('What do you want to add to Fat:') //Fat input`
`fat = food_add_fat_user_input + 'g' //Add g to the end of the string to repesent grams`

**Reply by JalonSolov** [reply]
`os.input` gives you a string, so what you have should work.

If you had an int:
```
i := 12
mut a := i.str()
println(a)
a = '$a$i'
println(a)
```
Output is:
```
12
1212
```
... in other words, you can use the `.str()` method on a numeric value, or you can use string interpolation.  You can also add strings, etc., etc.  Lots of ways.

**Reply by MrMidnightOfficial** [reply]
> `os.input` gives you a string, so what you have should work.
> 
> If you had an int:
> 
> ```
> i := 12
> mut a := i.str()
> println(a)
> a = '$a$i'
> println(a)
> ```
> 
> Output is:
> 
> ```
> 12
> 1212
> ```
> 
> ... in other words, you can use the `.str()` method on a numeric value, or you can use string interpolation. You can also add strings, etc., etc. Lots of ways.

TYSM you have been a great help
